reader-osm/src/main/java/com/graphhopper/reader/osm/OSMReader.java

```diff
@@ -859,7 +863,25 @@ private long createNewNodeId() {
      * @return the OSM turn relation, <code>null</code>, if unsupported turn relation
      */
     OSMTurnRelation createTurnRelation(ReaderRelation relation) {
-        OSMTurnRelation.Type type = OSMTurnRelation.Type.getRestrictionType(relation.getTag("restriction"));
+        String tagRestriction = null;
+        String vehicleTypeRestricted = null;
+        List<String> vehicleTypesExcept = null;
+        if (relation.hasTag("restriction")) {
+            tagRestriction = relation.getTag("restriction");
+        } else if (relation.hasTagStartsWith("restriction:")) {
+            vehicleTypeRestricted = relation.getKeyStartsWith("restriction:");
+            if (!Helper.isEmpty(vehicleTypeRestricted)) {
+                tagRestriction = relation.getTag(vehicleTypeRestricted);
```

Comment:
this one I do not understand. we know relation has a tag that startss with "restriction:", but how can we know there is a tag for the key `vehicleTypeRestricted` ? if there is not we get a null pointer exception ?

Reply:
I added a function `getKeyStartsWith` in `ReaderElement`, to get the key (not the value) starting with "restriction:", for example, we will have `vehicleTypeRestricted` = "restriction:motor_vehicle". Then it is parsed (removing "restriction:") `vehicleTypeRestricted`  = 
"motor_vehicle" to be stored in OSMTurnRelation.